lib/kafka/compressor.rb

```diff
@@ -0,0 +1,34 @@
+require "kafka/compression"
+
+module Kafka
+
+  # Compresses message sets using a specified codec.
+  #
+  # A message set is only compressed if its size meets the defined threshold.
+  class Compressor
+
+    # @param codec_name [Symbol, nil]
+    # @param threshold [Integer] the minimum number of messages in a message set
+    #   that will trigger compression.
+    def initialize(codec_name:, threshold:)
+      @codec = Compression.find_codec(codec_name)
+      @threshold = threshold
+    end
+
+    # @param message_set [Protocol::MessageSet]
+    # @return [Protocol::MessageSet]
+    def compress(message_set)
+      return message_set if @codec.nil? || message_set.size < @threshold
```

Comment:
Because `message_set` is now passed in as a param. I could have `compress?(message_set)`, but it wouldn't be quite as nice.